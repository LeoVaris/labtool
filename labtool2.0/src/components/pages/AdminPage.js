import React, { useEffect } from 'react'
import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { Button, Loader, Label, Table, Popup } from 'semantic-ui-react'
import { Link } from 'react-router-dom'
import { Redirect } from 'react-router'
import { HorizontalScrollable } from '../HorizontalScrollable'
import { formatCourseName } from '../../util/format'
import { getAllCI } from '../../services/courseInstance'
import { resetLoading } from '../../reducers/loadingReducer'
import { getAllUsers, updateOtherUser } from '../../services/user'
import { clearNotifications } from '../../reducers/notificationReducer'
import { usePersistedState } from '../../hooks/persistedState'
import { sortUsersBySysopLastname } from '../../util/sort'

const AdminCourseList = ({ courses }) => {
  if (!Array.isArray(courses)) {
    return <div />
  }

  return (
    <HorizontalScrollable>
      <Table singleLine color="yellow" style={{ overflowX: 'visible' }}>
        <Table.Header>
          <Table.Row>
            <Table.HeaderCell colSpan="1">Course ID</Table.HeaderCell>
            <Table.HeaderCell colSpan="1">Course name</Table.HeaderCell>
            <Table.HeaderCell colSpan="1">Course start date</Table.HeaderCell>
            <Table.HeaderCell colSpan="2"> </Table.HeaderCell>
          </Table.Row>
        </Table.Header>

        <Table.Body>
          {courses.map(instance => (
            <Table.Row key={instance.id}>
              <Table.Cell>{instance.shorterId} </Table.Cell>
              <Table.Cell>
                <strong>{formatCourseName(instance.name, instance.ohid, instance.start)}</strong>
              </Table.Cell>

              <Table.Cell> {instance.europeanStart} </Table.Cell>
              <Table.Cell textAlign="center">
                <Popup
                  trigger={
                    <Button
                      circular
                      size="tiny"
                      icon={{ name: 'edit', size: 'large', color: 'orange' }}
                      as={Link}
                      to={{
                        pathname: `/labtool/ModifyCourseInstanceStaff/${instance.ohid}`,
                        state: { fromAdmin: true }
                      }}
                    />
                  }
                  position="top right"
                  content="Manage instructors for this course"
                />
              </Table.Cell>
            </Table.Row>
          ))}
        </Table.Body>
      </Table>
    </HorizontalScrollable>
  )
}

AdminCourseList.propTypes = {
  courses: PropTypes.oneOfType([PropTypes.array, PropTypes.object])
}

const AdminUserList = ({ me, users, updateOtherUser }) => {
  const setAdmin = (userId, name, newAdmin) => () => {
    const message = `Are you sure you want to ${newAdmin ? 'grant admin rights to' : 'remove admin rights from'} ${name}?`
    if (window.confirm(message)) {
      const selfInflicted = userId === me.user.id
      if (selfInflicted && !newAdmin && !window.confirm('You probably do NOT want to remove admin rights from yourself. Are you really sure?')) {
        return
      }
      updateOtherUser({ id: userId, sysop: newAdmin })
    }
  }
  const sortedUsers = sortUsersBySysopLastname(users)

  return (
    <Table singleLine color="yellow" fixed>
      <Table.Header>
        <Table.Row>
          <Table.HeaderCell>Name</Table.HeaderCell>
          <Table.HeaderCell>Status</Table.HeaderCell>
        </Table.Row>
      </Table.Header>
      <Table.Body>
        {sortedUsers.map(user => (
          <Table.Row key={user.id}>
            <Table.Cell>
              {user.firsts} {user.lastname}
              <br />
              <a href={`mailto:${user.email}`}>{user.email}</a>
            </Table.Cell>
            <Table.Cell>
              {user.teacher && (
                <Label color="orange" horizontal>
                  Teacher
                </Label>
              )}
              {user.sysop ? (
                <div>
                  <Label color="orange" horizontal>
                    Admin
                  </Label>
                  <Button onClick={setAdmin(user.id, `${user.firsts} ${user.lastname}`, false)} size="tiny" color="green">
                    Remove admin
                  </Button>
                </div>
              ) : (
                <div>
                  <Label horizontal>Not admin</Label>
                  <Button onClick={setAdmin(user.id, `${user.firsts} ${user.lastname}`, true)} size="tiny" color="green">
                    Make admin
                  </Button>
                </div>
              )}
            </Table.Cell>
          </Table.Row>
        ))}
      </Table.Body>
    </Table>
  )
}

AdminUserList.propTypes = {
  me: PropTypes.object.isRequired,
  users: PropTypes.array.isRequired,
  updateOtherUser: PropTypes.func.isRequired
}

export const AdminPage = props => {
  const state = usePersistedState('AdminPage', {
    coursesList: false,
    usersList: false
  })

  useEffect(() => {
    props.clearNotifications()
  }, [])

  useEffect(() => {
    if (state.coursesList) {
      props.resetLoading()
      props.getAllCI()
    }
  }, [state.coursesList])

  useEffect(() => {
    if (state.usersList) {
      props.resetLoading()
      props.getAllUsers()
    }
  }, [state.usersList])

  if (!props.user.user.sysop) {
    return <Redirect to={`/labtool`} />
  } else if (props.loading.loading) {
    return <Loader active />
  }

  return (
    <div className="AdminPage">
      <h2>Welcome to the admin interface.</h2>
      <Button compact className={`ui button`} onClick={() => (state.coursesList = !state.coursesList)}>
        {state.coursesList ? 'Hide courses' : 'Manage courses'}
      </Button>
      <br />
      <br />
      {state.coursesList && <AdminCourseList courses={props.courseInstance} />}
      <br />
      <hr />
      <br />
      <Button compact className={`ui button`} onClick={() => (state.usersList = !state.usersList)}>
        {state.usersList ? 'Hide users' : 'Manage users'}
      </Button>
      <br />
      {state.usersList && <AdminUserList me={props.user} users={props.users} updateOtherUser={props.updateOtherUser} />}
    </div>
  )
}

const mapStateToProps = state => {
  return {
    user: state.user,
    courseInstance: state.courseInstance,
    users: state.users,
    loading: state.loading
  }
}

const mapDispatchToProps = {
  getAllCI,
  resetLoading,
  getAllUsers,
  updateOtherUser,
  clearNotifications
}

AdminPage.propTypes = {
  user: PropTypes.object.isRequired,
  courseInstance: PropTypes.oneOfType([PropTypes.array, PropTypes.object]),
  users: PropTypes.array.isRequired,
  loading: PropTypes.object.isRequired,

  getAllCI: PropTypes.func.isRequired,
  resetLoading: PropTypes.func.isRequired,
  getAllUsers: PropTypes.func.isRequired,
  updateOtherUser: PropTypes.func.isRequired,
  clearNotifications: PropTypes.func.isRequired
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(AdminPage)
