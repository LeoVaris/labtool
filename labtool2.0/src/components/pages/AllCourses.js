import React, { Component } from 'react'
import { getAll } from '../../services/courseInstance'
import { Link } from 'react-router-dom'

class AllCourses extends Component {
  constructor(props) {
    super(props)
    this.state = {
      courses: [],
    }
  }

  componentDidMount() {
    getAll().then(courses =>
      this.setState({ courses })
    )
  }

  render() {
    return (
      <div>
        <ul>
          {
            this.state.courses.map(c => (
              <li key={c.id}>
                <Link to={`/courses/${c.id}`}>{c.name}</Link>
              </li>
            ))
          }
        </ul>
      </div>
    )
  }
}



export default AllCourses