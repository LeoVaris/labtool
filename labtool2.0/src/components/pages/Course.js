import React, { Component } from 'react'
import { getOne } from '../../services/courseInstance'



class Course extends React.Component {
  constructor(props) {
    super(props)
    this.state = {
      course: undefined,
    }
  }

  componentDidMount() {
    getOne(parseInt(this.props.match.params.number, 10)).then(
      course => this.setState({ course })
    )
  }

  render() {
    if (!this.state.course) {
      return <div>course not found!!!</div>
    }

    return (
      <div>
        <h1>{this.state.course.name} </h1>
      </div>
    )
  }
}

export default Course