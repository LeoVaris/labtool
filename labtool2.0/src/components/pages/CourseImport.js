import React, { Component } from 'react'
import { Button, Container, Header, Table, Loader, Form } from 'semantic-ui-react'
import { connect } from 'react-redux'
import { Link, Redirect } from 'react-router-dom'
import { getIsAllowedToImport, getImportableCourses, importCourses } from '../../services/courseImport'
import { resetLoading, addRedirectHook } from '../../reducers/loadingReducer'
import { HorizontalScrollable } from '../HorizontalScrollable'

/**
 *  Show all the courses in a single list.
 */
export class CourseImport extends Component {
  componentWillMount = async () => {
    await this.props.resetLoading()
    this.props.getIsAllowedToImport()
    this.props.getImportableCourses()
  }

  handleSubmit = async e => {
    e.preventDefault()
    const data = this.props.importable
    if (data) {
      const courses = data.filter(entry => e.target['import' + entry.hid] && e.target['import' + entry.hid].checked)
      this.props.addRedirectHook({
        hook: 'COURSE_IMPORT_DO_IMPORT_'
      })
      await this.props.importCourses({ courses })
    }
  }

  render() {
    return (
      <div className="CourseImport">
        <Container>
          <Header as="h2">Import courses</Header>
          {this.props.loading.loading || !Array.isArray(this.props.importable) || !this.props.canImport ? (
            <Loader active />
          ) : this.props.loading.redirect ? (
            <Redirect to="/labtool/courses" />
          ) : this.props.importable.length ? (
            <Form onSubmit={this.handleSubmit}>
              <HorizontalScrollable>
                <Table singleLine color="yellow" style={{ overflowX: 'visible' }}>
                  <Table.Header>
                    <Table.Row>
                      <Table.HeaderCell colSpan="1" key={-1}>
                        Import?
                      </Table.HeaderCell>
                      <Table.HeaderCell colSpan="1">Course id</Table.HeaderCell>
                      <Table.HeaderCell colSpan="1">Course name</Table.HeaderCell>
                      <Table.HeaderCell colSpan="1">Course start date</Table.HeaderCell>
                      <Table.HeaderCell colSpan="1">Course end date</Table.HeaderCell>
                    </Table.Row>
                  </Table.Header>

                  <Table.Body>
                    {this.props.importable.map(instance => (
                      <Table.Row key={instance.hid}>
                        <Table.Cell>
                          <Form.Checkbox name={'import' + instance.hid} />
                        </Table.Cell>
                        <Table.Cell>{instance.shorterId}</Table.Cell>
                        <Table.Cell>{instance.cname}</Table.Cell>
                        <Table.Cell>{instance.europeanStart}</Table.Cell>
                        <Table.Cell>{instance.europeanEnd}</Table.Cell>
                      </Table.Row>
                    ))}
                  </Table.Body>
                </Table>
              </HorizontalScrollable>

              <br />
              <br />

              <Button type="submit" className="ui green button" content="Import" />

              <br />
              <br />

              <Link to="/labtool/courses">
                <Button className="ui button" type="cancel">
                  Cancel
                </Button>
              </Link>
            </Form>
          ) : (
            <div>
              <h4>There are no courses in Kurki to import</h4>

              <Link to="/labtool/courses">
                <Button className="ui button" type="cancel">
                  Back
                </Button>
              </Link>
            </div>
          )}
        </Container>
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    loading: state.loading,
    canImport: state.courseImport.canImport,
    importable: state.courseImport.importable
  }
}

const mapDispatchToProps = {
  resetLoading,
  getIsAllowedToImport,
  getImportableCourses,
  importCourses,
  addRedirectHook
}

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(CourseImport)
