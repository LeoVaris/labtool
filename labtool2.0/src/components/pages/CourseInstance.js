import React from 'react'

const CourseInstance = ({ instance, handleFieldChange }) => {
  return(
    <li style={{ textAlignVertical: 'center', textAlign: 'center' }}> {instance.name} <button name={instance.name} onClick={handleFieldChange} value={instance.id}>Register</button>
    </li>
  )
}

export default CourseInstance