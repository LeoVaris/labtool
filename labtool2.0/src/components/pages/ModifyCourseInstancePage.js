import React, { Component } from 'react'
import { Form, Input, Button, Grid, Radio, Checkbox } from 'semantic-ui-react'
import { getOneCI, modifyOneCI } from '../../services/courseInstance'
import { setActive, setFinalReview } from '../../reducers/selectedInstanceReducer'
import { connect } from 'react-redux'
import { Link } from 'react-router-dom'
import { Redirect } from 'react-router'
import { clearNotifications } from '../../reducers/notificationReducer'

/**
 *  Page used to modify a courseinstances information. Can only be accessed by teachers.
 */
export class ModifyCourseInstancePage extends Component {
  componentWillMount() {
    this.props.clearNotifications()
    this.props.getOneCI(this.props.courseId)
  }

  componentDidUpdate() {
    if (this.props.notification.error !== undefined) {
      if (!this.props.notification.error && this.props.history) {
        this.props.history.push(`/labtool/courses/${this.props.selectedInstance.ohid}`)
      }
    }
  }

  changeActive = () => {
    const newValue = this.props.selectedInstance.active !== true
    this.props.setActive(newValue)
  }

  changeFinalReview = () => {
    const newValue = this.props.selectedInstance.finalReview !== true
    this.props.setFinalReview(newValue)
  }

  handleSubmit = async e => {
    try {
      e.preventDefault()

      const content = {
        weekAmount: e.target.weekAmount.value,
        weekMaxPoints: e.target.weeklyMaxpoints.value,
        currentWeek: e.target.currentWeek.value,
        active: this.props.selectedInstance.active,
        ohid: this.props.selectedInstance.ohid,
        finalReview: this.props.selectedInstance.finalReview
      }
      await this.props.modifyOneCI(content, this.props.selectedInstance.ohid)
      this.forceUpdate()
    } catch (error) {
      console.log(error)
    }
  }

  render() {
    if (this.props.redirect && this.props.redirect.redirect) {
      return <Redirect to={`/labtool/courses/${this.props.selectedInstance.ohid}`} />
    }
    const selectedInstance = { ...this.props.selectedInstance }
    return (
      <div className="CoursePage" style={{ textAlignVertical: 'center', textAlign: 'center' }}>
        <Grid>
          <Grid.Row centered>
            <h2> Edit course: {selectedInstance.name} </h2>
          </Grid.Row>
        </Grid>

        <Grid>
          <Grid.Row centered>
            {selectedInstance.active === true ? (
              <div>
                <Grid.Row>
                  <h4 style={{ color: '#21ba45' }}>This course is currently activated.</h4>
                </Grid.Row>
              </div>
            ) : (
              <div>
                <Grid.Row>
                  <h4 style={{ color: 'grey' }}>This course is currently passive.</h4>
                </Grid.Row>
              </div>
            )}
          </Grid.Row>
        </Grid>

        <Grid>
          <Grid.Row centered>
            <Form onSubmit={this.handleSubmit}>
              <Form.Group inline>
                <label style={{ width: '125px', textAlign: 'left' }}>Week amount</label>
                <Input name="weekAmount" required="true" type="text" style={{ maxWidth: '7em' }} defaultValue={JSON.stringify(selectedInstance.weekAmount)} className="form-control1" />
              </Form.Group>

              <Form.Group inline>
                <label style={{ width: '125px', textAlign: 'left' }}>Weekly maxpoints</label>
                <Input name="weeklyMaxpoints" required="true" type="text" style={{ maxWidth: '7em' }} defaultValue={JSON.stringify(selectedInstance.weekMaxPoints)} className="form-control2" />
              </Form.Group>

              <Form.Group inline>
                <label style={{ width: '125px', textAlign: 'left' }}>Current week</label>
                <Input name="currentWeek" required="true" type="text" style={{ maxWidth: '7em' }} defaultValue={JSON.stringify(selectedInstance.currentWeek)} className="form-control3" />
              </Form.Group>

              <Form.Group inline>
                <Checkbox
                  name="finalReview"
                  checked={this.props.selectedInstance.finalReview}
                  onChange={this.changeFinalReview}
                  label="Course has a final review"
                  style={{ width: '150px', textAlign: 'left' }}
                />
              </Form.Group>

              <Form.Group inline>
                <Checkbox name="courseActive" checked={this.props.selectedInstance.active} onChange={this.changeActive} label="Course is active" style={{ width: '150px', textAlign: 'left' }} />
              </Form.Group>

              <Form.Field>
                <Button type="Submit" floated="left" color="green" size="huge">
                  Save
                </Button>

                <Link to="/labtool/courses">
                  <Button type="Cancel" floated="right" color="red" size="huge">
                    Cancel
                  </Button>
                </Link>
              </Form.Field>
            </Form>
          </Grid.Row>
        </Grid>

        <Link to={`/labtool/ModifyCourseInstanceStaff/${this.props.selectedInstance.ohid}`}>
          <Button style={{ marginTop: '20px', marginLeft: '5px', marginRight: '5px' }} block="true">
            Add or remove assistant teachers
          </Button>
        </Link>
        <Link to={`/labtool/ModifyCourseInstanceCodeReviews/${this.props.selectedInstance.ohid}`}>
          <Button style={{ marginTop: '20px', marginLeft: '5px', marginRight: '5px' }} block="true">
            Add or modify codereviews
          </Button>
        </Link>
        <Link to={`/labtool/checklist/${this.props.selectedInstance.ohid}/create`}>
          <Button style={{ marginTop: '20px', marginLeft: '5px', marginRight: '5px' }} block="true">
            Create new checklist
          </Button>
        </Link>
        <Link to={`/labtool/managetags`}>
          <Button style={{ marginTop: '20px', marginLeft: '5px', marginRight: '5px' }} block="true">
            Edit tags
          </Button>
        </Link>
      </div>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return {
    selectedInstance: state.selectedInstance,
    notification: state.notification,
    redirect: state.redirect,
    ownProps
  }
}

export default connect(
  mapStateToProps,
  { getOneCI, modifyOneCI, clearNotifications, setActive, setFinalReview }
)(ModifyCourseInstancePage)
