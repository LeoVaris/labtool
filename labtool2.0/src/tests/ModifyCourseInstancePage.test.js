import React from 'react'
import { ModifyCourseInstancePage } from '../components/pages/ModifyCourseInstancePage'
import { shallow } from 'enzyme'
import { Popup, Dropdown, Checkbox } from 'semantic-ui-react'

describe('<ModifyCourseInstancePage />', () => {
  let wrapper

  const courseData = {
    id: 1,
    name: 'Aineopintojen harjoitustyÃ¶: Tietokantasovellus (periodi IV)',
    start: '2018-03-11T21:00:00.000Z',
    end: '2018-03-11T21:00:00.000Z',
    active: true,
    weekAmount: 7,
    weekMaxPoints: 2,
    currentWeek: 1,
    ohid: 'TKT20011.2018.K.A.1',
    currentCodeReview: [1, 2],
    amountOfCodeReviews: 3
  }

  const loading = {
    loading: false,
    loadingHooks: [],
    redirect: false,
    redirectHooks: [],
    redirectFailure: false
  }

  let mockFn = jest.fn()

  beforeEach(() => {
    wrapper = shallow(
      <ModifyCourseInstancePage
        codeReviewDropdowns={[{ value: 3, text: '3' }]}
        getOneCI={mockFn}
        clearNotifications={mockFn}
        loading={loading}
        resetLoading={mockFn}
        selectedInstance={courseData}
        courseId={''}
        notification={{}}
        redirect={{}}
        modifyOneCI={mockFn}
        changeCourseField={mockFn}
        addRedirectHook={mockFn}
        setFinalReview={mockFn}
      />
    )
  })

  describe('Modify Instance Component', () => {
    it('is ok', () => {
      true
    })

    it('should render without throwing an error', () => {
      expect(wrapper.find('.CoursePage').exists()).toEqual(true)
    })

    it('should render correctly', () => {
      expect(wrapper).toMatchSnapshot()
    })

    it('renders weekly amount', () => {
      expect(wrapper.find('.form-control1').length).toEqual(1)
    })

    it('renders weekly maxpoints', () => {
      expect(wrapper.find('.form-control2').length).toEqual(1)
    })
    it('renders current week', () => {
      expect(wrapper.find('.form-control3').length).toEqual(1)
    })
    it('renders currently visible code reviews', () => {
      expect(
        wrapper
          .find(Popup)
          .at(0)
          .props().trigger.props.value
      ).toEqual(1)
      expect(
        wrapper
          .find(Popup)
          .at(1)
          .props().trigger.props.value
      ).toEqual(2)
    })

    it('renders dropdown to show currently not visible code reviews', () => {
      expect(wrapper.find(Dropdown).length).toEqual(1)
      expect(wrapper.find(Dropdown).props().options[0].text).toEqual('3')
    })
    it('renders active course checkbox', () => {
      const checkbox = wrapper.find(Checkbox).find({ name: 'courseActive' })

      expect(checkbox.prop('label')).toEqual('Activate course')
    })
  })
})
