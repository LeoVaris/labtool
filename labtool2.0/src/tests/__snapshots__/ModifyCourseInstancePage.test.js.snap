// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModifyCourseInstancePage /> Modify Instance Component should render correctly 1`] = `
<div
  className="CoursePage"
  style={
    Object {
      "textAlign": "center",
      "textAlignVertical": "center",
    }
  }
>
  <Loader
    active={false}
    inline="centered"
  />
  <Grid>
    <GridRow
      centered={true}
    >
      <h2>
         Edit course: 
        Aineopintojen harjoitusty√∂: Tietokantasovellus (periodi IV)
         
      </h2>
    </GridRow>
  </Grid>
  <Grid>
    <GridRow
      centered={true}
    >
      <Form
        as="form"
        onSubmit={[Function]}
      >
        <FormGroup
          inline={true}
        >
          <label
            style={
              Object {
                "textAlign": "left",
                "width": "125px",
              }
            }
          >
            Week amount
          </label>
          <Input
            className="form-control1"
            name="weekAmount"
            onChange={[Function]}
            required="true"
            style={
              Object {
                "maxWidth": "7em",
              }
            }
            type="text"
            value={7}
          />
        </FormGroup>
        <FormGroup
          inline={true}
        >
          <label
            style={
              Object {
                "textAlign": "left",
                "width": "125px",
              }
            }
          >
            Weekly maxpoints
          </label>
          <Input
            className="form-control2"
            name="weekMaxPoints"
            onChange={[Function]}
            required="true"
            style={
              Object {
                "maxWidth": "7em",
              }
            }
            type="text"
            value={2}
          />
        </FormGroup>
        <FormGroup
          inline={true}
        >
          <label
            style={
              Object {
                "textAlign": "left",
                "width": "125px",
              }
            }
          >
            Current week
          </label>
          <Input
            className="form-control3"
            name="currentWeek"
            onChange={[Function]}
            required="true"
            style={
              Object {
                "maxWidth": "7em",
              }
            }
            type="text"
            value={1}
          />
        </FormGroup>
        <FormGroup
          inline={true}
        >
          <label
            style={
              Object {
                "textAlign": "left",
                "width": "125px",
              }
            }
          >
            Currently visible code reviews
          </label>
          <Popup
            content="Click to be removed on save"
            keepInViewPort={true}
            key="1"
            on="hover"
            position="top left"
            trigger={
              <Button
                as="button"
                compact={true}
                onClick={[Function]}
                role="button"
                value={1}
              >
                1
              </Button>
            }
          />
          <Popup
            content="Click to be removed on save"
            keepInViewPort={true}
            key="2"
            on="hover"
            position="top left"
            trigger={
              <Button
                as="button"
                compact={true}
                onClick={[Function]}
                role="button"
                value={2}
              >
                2
              </Button>
            }
          />
        </FormGroup>
        <FormGroup
          inline={true}
        >
          <Dropdown
            additionLabel="Add "
            additionPosition="top"
            closeOnBlur={true}
            deburr={false}
            fluid={true}
            icon="dropdown"
            minCharacters={1}
            multiple={true}
            noResultsMessage="No results found."
            onChange={[Function]}
            openOnFocus={true}
            options={
              Array [
                Object {
                  "text": "3",
                  "value": 3,
                },
              ]
            }
            placeholder="Select code review to set visible"
            renderLabel={[Function]}
            searchInput="text"
            selectOnBlur={true}
            selectOnNavigation={true}
            selection={true}
            wrapSelection={true}
          />
        </FormGroup>
        <FormGroup
          inline={true}
        >
          <Checkbox
            label="Course has a final review"
            name="finalReview"
            onChange={[Function]}
            style={
              Object {
                "textAlign": "left",
                "width": "150px",
              }
            }
            type="checkbox"
          />
        </FormGroup>
        <FormGroup
          inline={true}
        >
          <Checkbox
            checked={true}
            label="Activate course"
            name="courseActive"
            onChange={[Function]}
            style={
              Object {
                "textAlign": "left",
                "width": "150px",
              }
            }
            type="checkbox"
          />
        </FormGroup>
        <FormGroup>
          <Button
            as="button"
            color="green"
            floated="left"
            role="button"
            size="huge"
            type="Submit"
          >
            Save
          </Button>
          <Link
            replace={false}
            to="/labtool/courses"
          >
            <Button
              as="button"
              color="red"
              floated="right"
              role="button"
              size="huge"
              type="Cancel"
            >
              Cancel
            </Button>
          </Link>
        </FormGroup>
      </Form>
    </GridRow>
  </Grid>
  <Link
    replace={false}
    to="/labtool/ModifyCourseInstanceStaff/TKT20011.2018.K.A.1"
  >
    <Button
      as="button"
      block="true"
      role="button"
      style={
        Object {
          "marginLeft": "5px",
          "marginRight": "5px",
          "marginTop": "20px",
        }
      }
    >
      Add or remove assistant teachers
    </Button>
  </Link>
  <Link
    replace={false}
    to="/labtool/ModifyCourseInstanceCodeReviews/TKT20011.2018.K.A.1"
  >
    <Button
      as="button"
      block="true"
      role="button"
      style={
        Object {
          "marginLeft": "5px",
          "marginRight": "5px",
          "marginTop": "20px",
        }
      }
    >
      Add or modify codereviews
    </Button>
  </Link>
  <Link
    replace={false}
    to="/labtool/checklist/TKT20011.2018.K.A.1/create"
  >
    <Button
      as="button"
      block="true"
      role="button"
      style={
        Object {
          "marginLeft": "5px",
          "marginRight": "5px",
          "marginTop": "20px",
        }
      }
    >
      Create new checklist
    </Button>
  </Link>
  <Link
    replace={false}
    to="/labtool/managetags"
  >
    <Button
      as="button"
      block="true"
      role="button"
      style={
        Object {
          "marginLeft": "5px",
          "marginRight": "5px",
          "marginTop": "20px",
        }
      }
    >
      Edit tags
    </Button>
  </Link>
</div>
`;
